In [1]:
#My prefered Web scraping tool is Selenium.
import selenium

In [2]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys

In [3]:
ser = Service(r'D:\Desktop\ML\Projects\WebScraping+Analysis Internship Assignment\webdriver\chromedriver.exe')

In [4]:
driver = webdriver.Chrome(service=ser)

In [5]:
import os

In [6]:
import pandas as pd

In [7]:
df = pd.read_excel('Output Data Structure.xlsx')

In [8]:
df.shape

(170, 15)

In [9]:
df.head()

URL_ID                                                URL  POSITIVE SCORE  \
0     1.0  https://insights.blackcoffer.com/how-is-login-...             NaN   
1     2.0  https://insights.blackcoffer.com/how-does-ai-h...             NaN   
2     3.0  https://insights.blackcoffer.com/ai-and-its-im...             NaN   
3     4.0  https://insights.blackcoffer.com/how-do-deep-l...             NaN   
4     5.0  https://insights.blackcoffer.com/how-artificia...             NaN   

   NEGATIVE SCORE  POLARITY SCORE  SUBJECTIVITY SCORE  AVG SENTENCE LENGTH  \
0             NaN             NaN                 NaN                  NaN   
1             NaN             NaN                 NaN                  NaN   
2             NaN             NaN                 NaN                  NaN   
3             NaN             NaN                 NaN                  NaN   
4             NaN             NaN                 NaN                  NaN   

   PERCENTAGE OF COMPLEX WORDS  FOG INDEX  AVG NUMBER OF WORDS PER SENTENCE  \
0                          NaN        NaN                               NaN   
1                          NaN        NaN                               NaN   
2                          NaN        NaN                               NaN   
3                          NaN        NaN                               NaN   
4                          NaN        NaN                               NaN   

   COMPLEX WORD COUNT  WORD COUNT  SYLLABLE PER WORD  PERSONAL PRONOUNS  \
0                 NaN         NaN                NaN                NaN   
1                 NaN         NaN                NaN                NaN   
2                 NaN         NaN                NaN                NaN   
3                 NaN         NaN                NaN                NaN   
4                 NaN         NaN                NaN                NaN   

   AVG WORD LENGTH  
0              NaN  
1              NaN  
2              NaN  
3              NaN  
4              NaN

In [10]:
urls = df['URL'].tolist()

In [11]:
urls[0]

'https://insights.blackcoffer.com/how-is-login-logout-time-tracking-for-employees-in-office-done-by-ai/'

### Test Extraction with one link

In [12]:
url = urls[0]
driver.get(url)

In [13]:
search = driver.find_element(By.CLASS_NAME,"td-post-content")

In [14]:
search.text

'When people hear AI they often think about sentient robots and magic boxes. AI today is much more mundane and simple—but that doesn’t mean it’s not powerful. Another misconception is that high-profile research projects can be applied directly to any business situation. AI done right can create an extreme return on investments (ROIs)—for instance through automation or precise prediction. But it does take thought, time, and proper implementation. We have seen that success and value generated by AI projects are increased when there is a grounded understanding and expectation of what the technology can deliver from the C-suite down.\n“Artificial Intelligence (AI) is a science and a set of computational technologies that are inspired by—but typically operate quite differently from—the ways people use their nervous systems and bodies to sense, learn, reason and take action.”3 Lately there has been a big rise in the day-to-day use of machines powered by AI. These machines are wired using cro

### Next step:

For each of the articles, given in the input.xlsx file, extract the article text and save the extracted article in a text file with URL_ID as its file name

In [15]:
os.chdir('./articles')

In [16]:
# with open(f'{1}.txt', 'w') as f:
#     f.write(search.text)

## Final Text Extraction:

In [17]:
import sys

In [18]:
for i in range(len(urls)):
    sys.stdout.write('\r')
    sys.stdout.write("[%-20s] %d/%d" % ('='*(int(20*i/len(urls))) +'>', i+1, len(urls)))
    sys.stdout.flush()
    
    try:
        driver.get(urls[i])
        search = driver.find_element(By.CLASS_NAME,"td-post-content")
    except UnicodeEncodeError:
        print(f"Error in extracting {i}")
        search = ''
    
    with open(f'{i+1}.txt', 'w', encoding="utf-8") as f:
        f.write(search.text)

[===================>] 170/170

## All 170 text files have been successfully saved under the /articles folder.

In [19]:
driver.quit()